scripts/start.js

```diff
@@ -33,7 +34,8 @@ compiler.plugin('done', function (stats) {
 
     console.log(chalk.green('Compiled successfully!'));
     console.log('\nThe app is running at:');
-    console.log('\n    ' + chalk.cyan('http://localhost:' + port + '/'));
+    // TODO: display the correct port, if the default is busy.
+    console.log('\n    ' + chalk.cyan('http://localhost:' + DEFAULT_PORT + '/'));
```

Comment:
@halfzebra Thanks! You gave me new point of view on it.
My thoughts:
>Specify port from a flag -p or --port #36

It can be added easily on both variants.

> - Prompt user about occupied port, and ask for Y/n answer
(this is how it's done in create-react-app and I think it makes sense)
- If port is occupied and user answered Y to the prompt, auto-detect and use any available free port

Oh, ok. I don't think about that concerns. Thanks for sharing it. (And sorry, I have never used `create-react-app` before.  The time has come 😸 ).

>On top of that, all the code from start.js will be available to the user, so it must be as simple as it is possible.

Sure. But lots of closures and global vars with its reassignment is not so simple for user, in my point of view. We can think how it can be reimplemented in simpler manner.

> ...problems with promises, but I suspect, that some people might have.

...for users who knows Elm :neckbeard:. I think that promises is already a mature standard for writing code.

> Should we go full-promise in start.js

In case of that we want to use `detect` I think promises is more preferable than lots of closures and global vars. Will look on it one more time today, but later.

>and should we add a prompt?

I think that this feature tries to help(??) user some how but makes code a bit harder. When user will look at `start.js` it will see not so simple logic. And this case breaks `start.js will be available to the user, so it must be as simple as it is possible.`
And the last thing. Lets imagine: you start an app, then see the prompt... You don't think about ports, you don't know what port is available now to type it in prompt. All that you want is _run my pretty Elm application pleeeeease_ 🐱 . If you wanna change a port% please use `-p or --port`.